### ISSUE TITLE
install nightly `numpy` in the upstream-dev CI

### ISSUE THREAD

**Issue by keewis** [xarray-contrib/xdggs#44]
Our upstream-dev CI finally works for `healpy` (see #42), but `h3ronpy` and `pandas` make use of `pyarrow` for which I can't find wheels compatible with `numpy>=2.0`.

As a consequence, we can run the nightly tests with `numpy>=2.0`, and thus will only notice stuff breaking once `numpy=2.0` is out. It will also allow us to avoid seeing all the stuff that breaks for `pandas`, so not sure how urgent this is.

**Comment by keewis** [comment]
nightly wheels for `pyarrow` are at https://pypi.fury.io/arrow-nightlies. To install a `numpy>=2` compatible `pyarrow`, we can use
```sh
python -m pip install \
    -i https://pypi.fury.io/arrow-nightlies/ \
    --prefer-binary \
    --no-deps \
    --pre \
    --upgrade \
    pyarrow
```

**Comment by keewis** [comment]
closed by #51
